src/main/java/ch/njol/skript/bukkitutil/AdventureSoundReceiver.java

```diff
@@ -0,0 +1,104 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.bukkitutil;
+
+import java.util.Locale;
+import java.util.OptionalLong;
+
+import org.bukkit.Location;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Sound;
+import org.bukkit.SoundCategory;
+import org.bukkit.entity.Entity;
+import org.jetbrains.annotations.NotNull;
+
+import ch.njol.skript.effects.EffPlaySound;
+
+/**
+ * A utility interface to access the Player::playSound while also providing the same arguments to World::playSound
+ * Used in EffPlaySound. Separated due to static versioning.
+ */
+@FunctionalInterface
+public interface AdventureSoundReceiver<T, E> {
+
+	void play(
+		@NotNull T receiver, @NotNull E emitter, @NotNull String sound,
+		@NotNull SoundCategory category, float volume, float pitch
+	);
+
+	static <T, E> void play(
+		@NotNull AdventureEmitterSoundReceiver<T> adventureLocationReceiver,
+		@NotNull AdventureEntitySoundReceiver<T> adventureEmitterReceiver,
+		@NotNull T receiver, @NotNull E emitter, @NotNull String[] sounds,
+		@NotNull SoundCategory category, float volume, float pitch, OptionalLong seed
+	) {
+		for (String sound : sounds) {
+			NamespacedKey key = null;
+			try {
+				Sound enumSound = Sound.valueOf(sound.toUpperCase(Locale.ENGLISH));
+				key = enumSound.getKey();
+			} catch (IllegalArgumentException alternative) {
+				sound = sound.toLowerCase(Locale.ENGLISH);
+				if (!EffPlaySound.KEY_PATTERN.matcher(sound).matches())
+					continue;
+				key = NamespacedKey.minecraft(sound);
```

Comment:
Using an invalid sound name (e.g. "minecraft:entity.zombified_piglin.death") throws a IllegalArgumentException exception that needs to be handled by the play sound syntax.

Reply:
Where does it throw an IllegalArgumentException?

This only uses the key, and this is the JavaDoc of the playSound method
```
This function will fail silently if Location or Sound are null. No sound
will be heard by the players if their clients do not have the respective
sound for the value passed.
 ```

Reply:
`key = NamespacedKey.minecraft(sound);` this line throws the exception

Reply:
Only if users inputs invalid characters. You don't think it should error to the user stating invalid characters?

Reply:
I think that failing silently would be most consistent with the behavior when the sound doesn't exist, or for nearly everything else in Skript, yes. I would not be opposed to a more friendly error being printed, but I think that's something we'd need to discuss further.